lib/http-server.js

```diff
@@ -148,6 +148,31 @@ function HttpServer(options) {
     });
   }
 
+  // fix named like "%" https://github.com/http-party/http-server/issues/603
+  before.push(function (req, res) {
+    function ishex(ch) {
+      return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+    }
+    var ind = 0 , url = req.url;
+    while ((ind = url.indexOf('%',ind)) !== -1) {
```

Comment:
Instead of this complex while loop, could we use a simple regular expression to look for lone `%`'s?

Reply:
I prefer loop because  read easy . so, I rarely use regular expressions 。